### ISSUE TITLE
Question on queues and detailed documentation on RStream

### ISSUE THREAD

**Issue by philmtd** [redisson/redisson#3136]
Hi, I'm a fairly new user of Redisson and have a question regarding queues and streams in Redisson.

I'm evaluating whether I could/should use `RStream` or not but unfortunately the documentation and examples I found are very short and didn't come with much explanation of the used methods/parameters and the reasons why.

My use case is fairly simple and could be managed with a queue but I want to make sure an element pulled from the queue was processed by an `ack` or something similar to prevent data loss in case of a failure. 

I found this question on StackOverflow which is basically the same problem I have: https://stackoverflow.com/questions/63575219/redisson-working-queue-dequeu-strategies-on-processing-message-element-on 
But unfortunately it's basically a monologue by the person asking without a final answer or solution.

Basically I am not interested in features like consumer groups etc. and I'd already be happy if I could have something like in the following example:

```java
RQueue<E> q;
E element = q.poll(); // prevents other clients to poll the same element but does not finally remove from queue
processElement(element); // do something with it
q.ack(element); // acknowledges processing of element and removes element from queue
```

My questions are:
1. Would you recommend using `RStream` for this use case or is there a better way to achieve what I want with Redisson?
2. If I can use `RStream` for this is there a simple example to get `RStream` working like a `RQueue` with `ack`s?
2. Could you provide a more detailed documentation on `RStream`s?

Thank you!

**Comment by mrniko** [comment]
Yes, RStream object is the only object you can use for this task.

I think Reliable Topic object is exactly that you're looking for. https://github.com/redisson/redisson/issues/3131

**Comment by mrniko** [comment]
How many consumers do you have for the same list?

**Comment by philmtd** [comment]
Realistically I won't have more than a handful of consumers. Probably not more than 8 and pretty sure most of the time less than that.

**Comment by philmtd** [comment]
Oh by the way, what would be the correct way to create a consumer group in RStreams? When I just createGroup() I run into exceptions if it already exists. When I listGroups() on the stream to check whether the group already exists I run into an exception if the stream is not "created" ? yet. Should I just always createGroup() and try-catch around it?

Also for a scenario where one consumer fails to process a message (e.g. it crashes/gets killed), will the message automatically be "unlocked" and dispatched to the next consumer?

**Comment by mrniko** [comment]
> When I listGroups() on the stream to check whether the group already exists I run into an exception if the stream is not "created" ?

You can check its existence with `isExists()` method.

> Also for a scenario where one consumer fails to process a message (e.g. it crashes/gets killed), will the message automatically be "unlocked" and dispatched to the next consumer?

No, you need to use claim method.

**Comment by philmtd** [comment]
Thanks!

**Comment by philmtd** [comment]
I guess RStreams are too complicated for our use case, then. We don't want to have to track consumer names for recovery scenarios, we don't need consumer groups and we don't publish multiple fields/"keys" in a stream.
We will probably work around RQueue to get what we want as this seems easier than working with RStream. I'd love if Redisson would add a Queue with `ack()` in the future, though. I'm pretty sure there are more use cases out there that would benefit from a simple reliable queue (similar to my original example in this issue) without over-complicating it. :)

**Comment by mrniko** [comment]
Can you add feature request regarding this issue?

**Comment by philmtd** [comment]
Sure, done: #3157
